From 2b8b78d04a722b28a103ed1c6a63550c3e45244d Mon Sep 17 00:00:00 2001
From: Peter Varga <pvarga@inf.u-szeged.hu>
Date: Tue, 9 Sep 2014 11:04:49 +0200
Subject: [PATCH] Add --log-level switch

The --log-level switch sets the log level of the chromium messages. It affects
the browser and web processes but has no impact on the Qt messages.

The following levels can be set:
     0 -> LOG_INFO
     1 -> LOG_WARNING
     2 -> LOG_ERROR
     3 -> LOG_FATAL

Log level 3 set per default (fatal messages are shown only) since
lower level messages might be irrelevant for those who use QtWebEngine API.

Change-Id: Ie2ba10718d3c86bbfe3d847471b3facb346be446
Reviewed-by: Andras Becsi <andras.becsi@digia.com>
---
 src/core/content_main_delegate_qt.cpp | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/core/content_main_delegate_qt.cpp b/src/core/content_main_delegate_qt.cpp
index 10ee4fc..e418889 100644
--- a/src/core/content_main_delegate_qt.cpp
+++ b/src/core/content_main_delegate_qt.cpp
@@ -36,8 +36,12 @@
 
 #include "content_main_delegate_qt.h"
 
+#include "base/command_line.h"
+#include "base/logging.h"
 #include "base/path_service.h"
+#include "base/strings/string_number_conversions.h"
 #include "content/public/common/content_paths.h"
+#include "content/public/common/content_switches.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/ui_base_paths.h"
 #include "ui/base/resource/resource_bundle.h"
@@ -60,6 +64,19 @@ void ContentMainDelegateQt::PreSandboxStartup()
 {
     net::NetModule::SetResourceProvider(PlatformResourceProvider);
     ui::ResourceBundle::InitSharedInstanceWithLocale(l10n_util::GetApplicationLocale(std::string("en-US")), 0);
+
+    // Suppress info, warning and error messages per default.
+    int logLevel = logging::LOG_FATAL;
+
+    CommandLine* parsedCommandLine = CommandLine::ForCurrentProcess();
+    if (parsedCommandLine->HasSwitch(switches::kLoggingLevel)) {
+        std::string logLevelValue = parsedCommandLine->GetSwitchValueASCII(switches::kLoggingLevel);
+        int level = 0;
+        if (base::StringToInt(logLevelValue, &level) && level >= logging::LOG_INFO && level < logging::LOG_NUM_SEVERITIES)
+            logLevel = level;
+    }
+
+    logging::SetMinLogLevel(logLevel);
 }
 
 content::ContentBrowserClient *ContentMainDelegateQt::CreateContentBrowserClient()
-- 
2.7.4 (Apple Git-66)

